src/com/collection/DoublyLinkedList.java

```diff
@@ -0,0 +1,120 @@
+package com.collection;
+
+public class DoublyLinkedList {
+
+    Node head;
+    public Node getHead() { return head; }
+    public void setHead(Node head) { this.head = head; }
+
+
+    class Node {
+        ComparableStudent data;
+        Node previous;
+        Node next;
+
+        public Node(ComparableStudent data){
+            this(data, null, null);
+        }
+
+        public Node(ComparableStudent data, Node previous, Node next) {
+            this.data = data;
+            this.previous = previous;
+            this.next = next;
+        }
+
+        public ComparableStudent getData() { return data; }
+        public void setData(ComparableStudent data) { this.data = data; }
+        public Node getPrevious() { return previous; }
+        public void setPrevious(Node previous) { this.previous = previous; }
+        public Node getNext() { return next; }
+        public void setNext(Node next) { this.next = next; }
+    }
+
+    //inserts at the front
+    public void push(ComparableStudent student){
+
+        Node first = new Node(student);
```

Comment:
what if student is null?

Reply:
throw new IllegalArgumentException("Argument cannot be null");   ??